llama-rs/src/loader2.rs

```diff
@@ -62,7 +62,7 @@ pub(crate) fn load(
     ggml_loader::load_model_from_reader(&mut reader, &mut loader)
         .map_err(|err| LoadError::from_ggml_loader_error(err, path.clone()))?;
 
-    Ok(loader.model.expect("model should be initialized"))
+    loader.model.ok_or(LoadError::ModelNotCreated { path })
```

Comment:
the way I wrote the code, `Err` is actually impossible here.

Reply:
It's possible if there are no tensors at all in the model (e.g. `tensor_buffer` never gets called). Users should be aware of that.